File: tests/xcalablemp/NOT-IMPLEMENTED/async/C/reduction_async24.c
/*testp206.c*/
/*組込み手続きのテスト*/
#include<xmp.h>
#include<mpi.h>
#include<stdio.h>
#include<stdlib.h>
double time,tick,max_tick;
double xmp_wtime(),xmp_wtick();
int async_val;
char *result;     
int i;
int main(void){
   result = "OK";
   time = xmp_wtime();
   tick = xmp_wtick();
   max_tick = tick;

#pragma xmp reduction(max: tick) async(1)

   async_val = xmp_test_async(1);
   if(!async_val){
      async_val = xmp_test_async(1);
   }

#pragma xmp wait_async(1)
 
   if(tick != max_tick) result = "NG";

   printf("%d %s %s\n",xmp_node_num(),"testp206.c",result); 
   return 0;
}
      
         
      
   
   

File: tests/xcalablemp/not-implemented-yet/async/F/reduction_async24.f90
! testp206.f
! 組込み手続のテスト

      program main
      include 'xmp_lib.h'
      real*8 time, tick, max_tick
      logical async_val
      character(len=2) result

      result = 'OK'
      time = xmp_wtime()
      tick = xmp_wtick()
      max_tick = tick
      
!$xmp reduction(max: tick) async(1)

 10   continue
      async_val = xmp_test_async(1)
      if(.not.async_val) goto 10

!$xmp wait_async(1)

      if(tick .ne. max_tick) result = 'NG'

      print *, xmp_node_num(), 'testp206.f ', result

      end
      
      
